doc/dev/error-message.md

```diff
@@ -0,0 +1,20 @@
+# Error message
+
+Elektra has a lot of developers and hence the error messages vary in how they are written.
+This guideline should unify the message format so that users will get consistent error messages.
+
+- All error messages start with a capital letter. The only exception are quoted messages at the start such as "'xxxx-xxxx' is an invalid syntax".
+  If you reference a method name such as `kdbSet()` in the first word, surround it with single quotes.
+  Try to avoid variable strings at the start of the sentence ("%s"-strings) if possible because you cannot guarantee capital casing.
+- **Always** provide the errno if possible. If possible use `strerror_r` for writing out the message because it is thread safe.
```

Comment:
I like the guideline that errno is always passed to the user. But always using `strerror_r` is very tedious.

As errno should be only relevant for resource errors, you could simply have a macro for resource errors with file name and errno.